File: src/bin/decode_proto_bin_test.rs
use luwen_ref::detect_chips;

fn main() {
    let devices = detect_chips().unwrap();
    let bh = devices[0].as_bh();
    println!();
    if let Some(bh) = bh {
        // Test decoding the bin from the boot fs
        let decode_msg = bh.decode_boot_fs_table("boardcfg");
        println!("Decoded boardcfg: {:#?}", decode_msg);
        println!();

        let decode_msg = bh.decode_boot_fs_table("flshinfo");
        println!("Decoded flshinfo: {:#?}", decode_msg);
        println!();

        let decode_msg = bh.decode_boot_fs_table("cmfwcfg");
        println!("Decoded cmfwcfg: {:#?}", decode_msg);
        println!();

        let decode_msg = bh.decode_boot_fs_table("origcfg");
        println!("Decoded origcfg: {:#?}", decode_msg);
        println!();
    }
}

File: bind/pyluwen/python_unit_tests/decode_proto_bin_test.py
from pyluwen import detect_chips
from pprint import pprint

def main():
    devices = detect_chips()
    for i, dev in enumerate(devices):
        if dev.as_bh():
            boardcfg = dev.as_bh().decode_boot_fs_table("boardcfg")
            print("boardcfg decoded:")
            pprint(boardcfg)

            flshinfo = dev.as_bh().decode_boot_fs_table("flshinfo")
            print("flshinfo decoded:")
            pprint(flshinfo)

            cmfwcfg = dev.as_bh().decode_boot_fs_table("cmfwcfg")
            print("cmfwcfg decoded:")
            pprint(cmfwcfg)

            origcfg = dev.as_bh().decode_boot_fs_table("origcfg")
            print("origcfg decoded:")
            pprint(origcfg)

        else:
            print("Nothing to do for non BH chips")

if __name__ == "__main__":
    main()
